Platform/ApollolakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -670,27 +689,12 @@ BoardInit (
     RegisterHeciService();
     InitPlatformService ();
 
-    // Lock down SPI for all other payload entry except FWUpdate and OSloader
-    // as this phase is too early for them to lock it here
-    SpiBaseAddress = GetDeviceAddr (OsBootDeviceSpi, 0);
-    SpiBaseAddress = TO_MM_PCI_ADDRESS (SpiBaseAddress);
-
-    if ((GetBootMode() != BOOT_ON_FLASH_UPDATE) && (GetPayloadId() != 0) && (GetPayloadId() != UEFI_PAYLOAD_ID_SIGNATURE)) {
-      // Set the BIOS Lock Enable and EISS bits
-      MmioOr8 (SpiBaseAddress + R_SPI_BCR, (UINT8) (B_SPI_BCR_BLE | B_SPI_BCR_EISS));
-
-      ClearFspHob ();
+    if ((GetBootMode() != BOOT_ON_FLASH_UPDATE) && (GetPayloadId() != 0)) {
+      ProgramSecuritySetting ();
     }
     break;
   case ReadyToBoot:
-    // Lock down SPI for everything
-    SpiBaseAddress = GetDeviceAddr (OsBootDeviceSpi, 0);
-    SpiBaseAddress = TO_MM_PCI_ADDRESS (SpiBaseAddress);
-
-    // Set the BIOS Lock Enable and EISS bits
-    MmioOr8 (SpiBaseAddress + R_SPI_BCR, (UINT8) (B_SPI_BCR_BLE | B_SPI_BCR_EISS));
-
-    ClearFspHob ();
+    ProgramSecuritySetting ();
```

Comment:
Can we make sure  ProgramSecuritySetting will be executed only once during the boot ?
Some MSR can only be set once.

Reply:
ProgramSecuritySetting () will be skipped for Osloader in EndOfStages, and be invoked in ReadyToBoot()
For other external Payloads,  ProgramSecuritySetting () will be invoked in ReadyToBoot().
So for all Payloads, ProgramSecuritySetting are expected to be executed once.
and we don't have MSR setting currently. we could update this if we really have other cases.

Reply:
Added payloadId check to make sure it only be invoked once.